sunpy/map/maputils.py

```diff
@@ -1,7 +1,259 @@
+    """
+    # Calculate all the edge pixels
+    nx, ny = smap.dimensions.x.value, smap.dimensions.y.value
+    bottom = list(product([0.0], np.arange(nx))) * u.pix
+    top = list(product([ny - 1], np.arange(nx))) * u.pix
+    lhs = list(product(np.arange(ny), [0])) * u.pix
+    rhs = list(product(np.arange(ny), [nx - 1])) * u.pix
+    return {"top": top, "bottom": bottom, "lhs": lhs, "rhs": rhs}
+
+
+def contains_full_disk(smap):
+    """
+    Checks if a map contains the full disk of the Sun.
+
+    A map contains the full disk of the Sun if the following two
+    conditions are met: (1) all the pixels at the edge of the map are
+    more than 1 solar radius from the center of the Sun and, (2) the
+    map is not all off disk. If both these conditions are met, the
+    function returns `True`. Otherwise, the function returns `False`.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        A map in helioprojective Cartesian coordinates.
+
+    Returns
+    -------
+    `~bool`
+        Returns `False` if any of the edge pixels are less than one solar
+        radius away from the center of the Sun.
+
+    Notes
+    -----
+    * The function assumes that the input map is rectangular.
+    * In the case of coronagraph images the disk itself need not be observed.
+
+    """
+    # Calculate all the edge pixels
+    edges = map_edges(smap)
+    edge_pixels = list(chain.from_iterable([edges["lhs"], edges["rhs"], edges["top"], edges["bottom"]]))
+    x = [p[0] for p in edge_pixels] * u.pix
+    y = [p[1] for p in edge_pixels] * u.pix
+
+    # Calculate the edge of the world
+    edge_of_world = smap.pixel_to_world(x, y)
+
+    # Calculate the distance of the edge of the world in solar radii
+    distance = u.R_sun * np.sqrt(edge_of_world.Tx ** 2 + edge_of_world.Ty ** 2) / smap.rsun_obs
+
+    # Test if all the edge pixels are more than one solar radius distant
+    # and that the whole map is not all off disk.
+    return np.all(distance > 1*u.R_sun) and ~is_all_off_disk(smap)
+
+
+@u.quantity_input
+def coordinate_is_on_disk(coordinate, scale: u.arcsecond):
+    """
+    Checks if the helioprojective Cartesian coordinate is on disk.
+
+    The check is performed by comparing the coordinate's distance
+    from the center of the Sun to the solar radius.
+
+    Parameters
+    ----------
+    coordinate : `~astropy.coordinates.SkyCoord`, `~sunpy.coordinates.frames.Helioprojective`
+        The input coordinate. The coordinate frame must be
+        `~sunpy.coordinates.Helioprojective`.
+
+    scale : `~astropy.units.Quantity`
+        The pixel scale size in units of arcseconds.
+
+    Returns
+    -------
+    `~bool`
+        Returns `True` if the coordinate is on disk, `False` otherwise.
+    """
+    # Calculate the radii of every pixel in helioprojective Cartesian
+    # co-ordinate distance units.
+    return u.R_sun * (np.sqrt(coordinate.Tx ** 2 + coordinate.Ty ** 2) / scale) < 1 * u.R_sun
```

Comment:
R_sun is just an average. Don't you want this to work as a function of time?

Reply:
Isn't `u.R_sun` the radius of the sun?  It should be a constant.

Reply:
Oh sorry! I see. You are converting R_sun into arcsec using the plate_scale. I think a better and less confusing approach would be to use the observer location which would set the size of the Sun. I understand that image plate scale is a proxy for that but it also involves the pixel size which has nothing to do with this problem.

Reply:
I think @ehsteve is right, I think to do this accurately you need to use the observer distance and the `coordinate.rsun` attribute to calculate the angular size.

Reply:
oh well actually, pixel scale might be sufficient but you should use `coordinate.rsun` at the very least (SDO headers have a different rsun than `i.R_sun`).

Reply:
I changed how `coordinate_is_on_disk` works to eliminate the use of scale keyword.